Sources/ExtensionDictionary.swift

```diff
@@ -24,8 +24,50 @@
 //
 
 import Foundation
-
-internal extension Dictionary {
+public extension Dictionary {
+    // MARK: - Public Functions
+    /**
+     Retrieves value from dictionary given a key path delimited with
+     provided delimiter to indicate a nested value.
+     
+     For example, a dictionary with [ "outer" : [ "inner" : "value" ] ]
+     could retrive 'value' via  path "outer.inner", given a
+     delimiter of ''.
+     
+     - parameter keyPath:   Key path delimited by delimiter.
+     - parameter delimiter: Delimiter.
+     
+     - returns: Value retrieved from dic
+     */
+    public func valueForKeyPath(keyPath: String, withDelimiter delimiter: String = GlossKeyPathDelimiter) -> Any? {
+        let keys = keyPath.components(separatedBy: delimiter)
+        
+        guard keys.first as? Key != nil else {
+            print("[Gloss] Unable to use keyPath '\(keyPath)' as key on type: \(Key.self)")
+            return nil
+        }
+        
+        return self.findValue(keys: keys)
+    }
+    
+    /**
+     Retrieves value from dictionary given a key path delimited with
+     provided delimiter by going down the dictionary stack tree
+     
+     - parameter keys: Array of keys splited by delimiter
+     - parameter depthLevel: Indicates current depth level in the dictionary tree
+     - returns: object retrieved from dic
+     */
+    private func findValue(keys: [String], depthLevel: Int = 0) -> Any? {
```

Comment:
to be consistent, can you move this `private` function under a separate `MARK: -`